zeus/api/resources/repository_tests_by_job.py

```diff
@@ -0,0 +1,61 @@
+from flask import current_app
+
+from zeus.config import db
+from zeus.constants import Result, Status
+from zeus.models import Build, Repository, TestCase, Job, Source
+
+from .base_repository import BaseRepositoryResource
+from ..schemas import TestCaseSummarySchema
+
+testcases_schema = TestCaseSummarySchema(many=True, strict=True)
+
+
+class RepositoryTestsByJobResource(BaseRepositoryResource):
+    def get(self, repo: Repository):
+        """
+        Return a list of testcases for the given repository.
+        """
+        # use the most recent successful build to fetch test results
+        builds = (
+            db.session.query(Build.id)
+            .join(Source, Source.id == Build.source_id)
+            .filter(
+                Source.patch_id == None,  # NOQA
+                Build.repository_id == repo.id,
+                Build.result == Result.passed,
+                Build.status == Status.finished,
+            )
+            .order_by(Build.number.asc())
+            .limit(20)
+        )
+
+        if not builds:
+            current_app.logger.info("no successful builds found for repository")
+            return self.respond([])
+
+        job_list = db.session.query(Job.id).filter(Job.build_id.in_(builds)).limit(30)
```

Comment:
oh this will probably not work out well in some scenarios

Reply:
do you need full jobs? if you could pair it down the API needs it'd be easy to return e.g. 1000 jobs

Reply:
Probably don't need full jobs